test/handlers/redeem_and_unstake/RedeemIntentConfirmedHandler/persist.test.ts

```diff
@@ -0,0 +1,129 @@
+      contractAddress: '0x0000000000000000000000000000000000000001',
+    }];
+    saveStub = sinon.stub();
+    existingMessageRecord = StubData.messageAttributes(
+      transactions[0]._messageHash,
+      transactions[0].contractAddress,
+      new BigNumber('1'),
+    );
+    existingMessageRecord.type = type;
+    existingMessageRecord.direction = direction;
+
+    sinonMessageRepositoryMock = sinon.createStubInstance(MessageRepository, {
+      save: saveStub,
+      get: Promise.resolve(existingMessageRecord),
+    });
+
+    outputMessage = StubData.messageAttributes(
+      transactions[0]._messageHash,
+      transactions[0].contractAddress,
+      new BigNumber('1'),
+    );
+    outputMessage.targetStatus = MessageStatus.Declared;
+    outputMessage.type = type;
+    outputMessage.direction = direction;
+  });
+
+  it('should persist successfully when target status is undeclared', async (): Promise<void> => {
+    existingMessageRecord.targetStatus = MessageStatus.Undeclared;
+
+    const handler = new RedeemntentConfirmedHandler(sinonMessageRepositoryMock);
+    const models = await handler.persist(transactions);
+
+    assert.equal(
+      models.length,
+      transactions.length,
+      'Number of models must be equal to transactions',
+    );
+
+    assert.deepStrictEqual(models[0], outputMessage);
+    SpyAssert.assert(saveStub, 1, [[outputMessage]]);
+  });
+
+  it('should persist successfully when target status is undefined', async (): Promise<void> => {
+    existingMessageRecord.targetStatus = undefined as unknown as MessageStatus;
+
+    const handler = new RedeemntentConfirmedHandler(sinonMessageRepositoryMock);
+
+    const models = await handler.persist(transactions);
+
+    assert.equal(
+      models.length,
+      transactions.length,
+      'Number of models must be equal to transactions',
+    );
+
+    assert.deepStrictEqual(models[0], outputMessage);
+    SpyAssert.assert(saveStub, 1, [[outputMessage]]);
+  });
+
+  it('should create new entry when message is not present', async (): Promise<void> => {
+    const save = sinon.stub();
+
+    const mockMessageRepo = sinon.createStubInstance(MessageRepository, {
+      save: save as any,
+      get: Promise.resolve(null),
+    });
+    const handler = new RedeemntentConfirmedHandler(mockMessageRepo as any);
+
+    const models = await handler.persist(transactions);
+
+    const outputMessage = new Message(
+      transactions[0]._messageHash,
+      type,
+      transactions[0].contractAddress,
+      undefined,
+      MessageStatus.Declared,
+      undefined,
+      undefined,
+      new BigNumber(transactions[0]._redeemerNonce),
+      utils.toChecksumAddress(transactions[0]._redeemer),
+      direction,
+      undefined,
+      undefined,
+      transactions[0]._hashLock,
+      undefined,
+      undefined,
+    );
+
+    assert.equal(
+      models.length,
+      transactions.length,
+      'Number of models must be equal to transaction',
+    );
+
```

Comment:
may be also add a test case for RevocationDeclared

Reply:
Done!